x-pack/legacy/plugins/infra/public/components/logging/log_entry_flyout/log_entry_actions_menu.tsx

```diff
@@ -91,3 +105,18 @@ const getUptimeLink = (logItem: InfraLogItem) => {
     hash: `/?search=(${searchExpressions.join(' OR ')})`,
   });
 };
+
+const getAPMLink = (logItem: InfraLogItem) => {
+  const traceIdEntry = logItem.fields.find(
+    ({ field, value }) => value != null && field === 'trace.id'
+  );
+
+  if (!traceIdEntry) {
+    return undefined;
+  }
+
+  return url.format({
+    pathname: chrome.addBasePath('/app/apm'),
+    hash: `/traces?kuery=${encodeURIComponent(`trace.id:${traceIdEntry.value}`)}`,
```

Comment:
I think for now we want to add `rangeFrom` and `rangeTo` values here, because if a log happened 30 hours ago and you go to find that trace without specifying the date range, APM will default to "last 24 hours" and the query that we are pre-populating will find 0 results.

We can be fancy and calculate a range from the timestamp, or we can do `rangeFrom=now-1y&rangeTo=now` since we're only looking for a single trace. Thoughts?

Reply:
I tried the fancy version with the simple range as a failsafe.